File: app/src/main/java/org/xdty/callerinfo/di/MainComponent.java
package org.xdty.callerinfo.di;

import org.xdty.callerinfo.activity.MainActivity;
import org.xdty.callerinfo.di.modules.AppModule;
import org.xdty.callerinfo.di.modules.MainModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = { MainModule.class, AppModule.class })
public interface MainComponent {
    void inject(MainActivity view);
}

File: app/src/main/java/org/xdty/callerinfo/di/MainComponent.kt
package org.xdty.callerinfo.di

import dagger.Component
import org.xdty.callerinfo.activity.MainActivity
import org.xdty.callerinfo.di.modules.AppModule
import org.xdty.callerinfo.di.modules.MainModule
import javax.inject.Singleton

@Singleton
@Component(modules = [MainModule::class, AppModule::class])
interface MainComponent {
    fun inject(view: MainActivity)
}